### ISSUE TITLE
Creating a product does not work as documented, need to set variant field

### ISSUE THREAD

**Issue by poolebu** [Shopify/shopify_python_api#386]
Tried to create a product as documented, gives the following error

![error_001](https://user-images.githubusercontent.com/4500001/82441226-ea51e500-9a9d-11ea-8bed-525302d6c142.png)

Adding variant array works however

![works](https://user-images.githubusercontent.com/4500001/82441261-f9389780-9a9d-11ea-9554-e7346626d6f9.png)

**Comment by prysie** [comment]
![image](https://user-images.githubusercontent.com/12985914/86095001-9ef71180-baf4-11ea-8a73-bafd21e99098.png)

There is a check in the Product save function for API versions >= 201910 it attempts to read the variant attribute, however it has not been set to any value as it is not a required field as per the API specification.

**Comment by beetree** [comment]
I'm having a related issue using version 8.0.1. It can very easily be reproduced, and I've included full code below.

The following code...
```
import shopify
import ssl

ssl._create_default_https_context = ssl._create_unverified_context

keys = {
	"API_KEY": "XXX",
	"PASSWORD": "XXX"
}

shop_url = "https://%s:%s@XXX.myshopify.com/admin/api/2019-10" % (keys["API_KEY"], keys["PASSWORD"])
shopify.ShopifyResource.set_site(shop_url)

shopifyProduct = shopify.Product.find(XXX)
shopifyProduct.save()
```
... runs fine.

Whereas this code (only change is API version)...
```
import shopify
import ssl

ssl._create_default_https_context = ssl._create_unverified_context

keys = {
	"API_KEY": "XXX",
	"PASSWORD": "XXX"
}

shop_url = "https://%s:%s@XXX.myshopify.com/admin/api/2020-01" % (keys["API_KEY"], keys["PASSWORD"])
shopify.ShopifyResource.set_site(shop_url)

shopifyProduct = shopify.Product.find(XXX)
shopifyProduct.save()
```
... throws the following error:
```
Traceback (most recent call last):
  File "/Users/johan.land/PycharmProjects/FarmFoods/bin/tmp.py", line 15, in <module>
    shopifyProduct.save()
  File "/Users/johan.land/PycharmProjects/FarmFoods/venv/lib/python3.8/site-packages/shopify/resources/product.py", line 39, in save
    if 'variants' in self.product.attributes:
  File "/Users/johan.land/PycharmProjects/FarmFoods/venv/lib/python3.8/site-packages/pyactiveresource/activeresource.py", line 917, in __getattr__
    raise AttributeError(name)
AttributeError: product
```

I believe @prysie is right about where the issue is, but it seems that the loading of product does not set the variants correctly. As such, the behavior is not as one would expect (e.g. loading and then saving a product should not throw an error). Please note that @poolebu 's fix of adding setting the variants (`.variants=[]`) does not resolve the issue in my case.

**Comment by mbeasley32** [comment]
I'm having this same issue. I cannot create or update a product with product.save() at all. It throws the same error @beetree mentions:

Code: 
13: product = shopify.Product.find(5460519813285)
14: product.handle = "new handle"
15: product.save()


Error:
File "/home/mbeasley32/germrip_shopify_api.py", line 15, in <module>
    product.save()
  File "/home/mbeasley32/.local/lib/python3.8/site-packages/shopify/resources/product.py", line 39, in save
    if 'variants' in self.product.attributes:
  File "/home/mbeasley32/.local/lib/python3.8/site-packages/pyactiveresource/activeresource.py", line 917, in __getattr__
    raise AttributeError(name)
AttributeError: product

**Comment by poolebu** [comment]
Hello! So I'm back working with the same code, and the issue is still here but in another form. The default stable package at the moment, `8.0.1` does not let me create a product as documented. See bellow

```
API_VERSION = '2020-07'

```

![2020-08-01_13h06_05](https://user-images.githubusercontent.com/4500001/89100549-0c4dd900-d3f8-11ea-8bf7-717ef9d6f1ad.png)

![2020-08-01_13h05_14](https://user-images.githubusercontent.com/4500001/89100554-18399b00-d3f8-11ea-9c08-2a43ee13fdd1.png)

If I go back to version `!pip install ShopifyAPI==8.0.0` and `API_VERSION = '2020-04'`  I get the original error, which I can bypass by adding `new_product.variants = []` before saving a new product.

It seems to me that you should test the documented code for creating a new product before releasing a new api version, since its a critical blocker use case, or maybe I'm doing something wrong?

I hope this report is useful, and thank you for the API

**Comment by andyw8** [comment]
Hi @beetree and @poolebu 

Sorry to hear about these issues concerning problems with the library version and particular API versions.

We released v8.0.2 recently which should address this. Would you mind trying against the 2020-01 API (or later) that to see if it solves your issue?

We're also looking into ways to improve the release process to avoid similar issues in the future.

**Comment by beetree** [comment]
It looks like this is now working. At least the issue that previously caused my problems is resolved. I'll do the full upgrade now and see if I encounter any other bugs.

Thanks for fixing!
